Fim improvement (#2367)
wazuh/wazuh-documentation: source/user-manual/capabilities/file-integrity/fim-configuration.rst

Configuration
=============

> #. `Configuring syscheck - basic usage`_

Author: Done.

Configuration
=============

> After the installation, the Wazuh manager and the Wazuh agent come with the :ref:`default syscheck configuration <reference_ossec_syscheck_default_configuration>`, which should be reviewed and adjusted to the particular environment. This section presents different configuration examples, helping to learn how to use syscheck in different use cases:

Reviewer: What if we change `the Wazuh agent come with the ` for `the Wazuh agent have defined a `? Is this `come with` commonly used in this kind of documents?

Author: Done

> After the installation, the Wazuh manager and the Wazuh agent come with the :ref:`default syscheck configuration <reference_ossec_syscheck_default_configuration>`, which should be reviewed and adjusted to the particular environment. This section presents different configuration examples, helping to learn how to use syscheck in different use cases:

Reviewer: `This section presents different configuration examples, helping to learn how to use syscheck in different use cases` -> `This section presents configuration examples, helping to learn how to use it in different use cases`
How does this modification look for you? Avoiding the `different` repetition.

Author: Done

#. `Configuring who-data monitoring`_
#. `Configuring reporting file changes`_
#. `Configuring ignoring files and Windows registry entries`_
#. `Configuring ignoring files via rules`_
#. `Configuring the alert severity for the monitored files`_
#. `Configuring maximum recursion level allowed`_
#. `Configuring syscheck process priority`_
#. `Configuring where the database is to be stored`_
#. `Configuring synchronization`_

Syscheck component is configured both in the Wazuh manager's and in the Wazuh agent's :ref:`ossec.conf <reference_ossec_conf>` file. This capability can be also configured remotely using centralized configuration and the :ref:`agent.conf <reference_agent_conf>` file.
The list of all syscheck configuration options is available in the :ref:`syscheck <reference_ossec_syscheck>` section.

Configuring syscheck - basic usage
----------------------------------

To configure syscheck, a list of files and directories must be identified. The ``check_all`` attribute of the :ref:`directories <reference_ossec_syscheck_directories>` option allows checks of the file size, permissions, owner, last modification date, inode and all the hash sums (MD5, SHA1 and SHA256).

> By default, syscheck scans selected directories, whose list depends on the :ref:`default configuration <reference_ossec_syscheck_default_configuration>` for the host operating system.

Author: Done

.. warning:: There is a known bug that affects to the versions 2.8.5 and 2.8.4 of ``audit`` that shows a directory as ``null`` when it has been moved adding a ``/`` at the end of the directory. This bug will cause that no alerts related with this directory will be shown until a new event related to this directory is triggered when ``whodata`` is enabled.

How to tune audit to deal with a huge amount of who-data events at the same time
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

It is possible to lose ``who-data`` events when a flood of events appears. The following options help the audit socket and dispatcher to deal with big amounts of events:

> .. code-block:: xml

Reviewer 2: Exactly, the reason is that the lines 107 and 108 are not XML code, as specified in the code-block

  File '/test/private' modified
  Mode: real-time
  Changed attributes: size,mtime,inode,md5,sha1,sha256
  Size changed from '14' to '20'
  Old modification time was: '1585757413', now it is '1585757658'
  Old inode was: '23187', now it is '23185'
  Old md5sum was: 'ef4ad1a40d0a95ad2e1b72eccdca6d44'
  New md5sum is : '158ccd88359654ac4ffd0e3cecb79a49'
  Old sha1sum was: '5f34d30f7bdefe9e825bff388de047dacdc09853'
  New sha1sum is : '18dfef68273c00fc733e28ce9aa1830f5e8fabd8'
  Old sha256sum was: '211ae95d4e54cff5724a98f0bae0b505adfdafe1ed8b15e40570a5fe58d20c61'
  New sha256sum is : '60c2a08e66f02bacea882f7b437f9c983431d75a686b703661c34e288d36de9d'Attributes:
  - Size: 20
  - Permissions: rw-r--r--
  - Date: Wed Apr 1 16:14:18 2020
  - Inode: 23185
  - User: root (0)
  - Group: root (0)
  - MD5: 158ccd88359654ac4ffd0e3cecb79a49
  - SHA1: 18dfef68273c00fc733e28ce9aa1830f5e8fabd8
  - SHA256: 60c2a08e66f02bacea882f7b437f9c983431d75a686b703661c34e288d36de9dWhat changed:
  <Diff truncated because nodiff option>

.. _how_to_fim_ignore:

Configuring ignoring files and Windows registry entries
-------------------------------------------------------

> In order to avoid false positives, syscheck can be configured to ignore certain files and directories that don’t need to be monitored by using the :ref:`ignore <reference_ossec_syscheck_ignore>` option:

Reviewer 3: ```suggestion
In order to avoid false positives, syscheck can be configured to ignore certain files and directories that do not need to be monitored by using the :ref:`ignore <reference_ossec_syscheck_ignore>` option:
```

    <ignore type="sregex">.log$|.tmp</ignore>
  </syscheck>

Similar functionality, but for the Windows registries can be achieved by using the :ref:`registry_ignore <reference_ossec_syscheck_registry_ignore>` option:

.. code-block:: xml

  <syscheck>
   <registry_ignore>HKEY_LOCAL_MACHINE\Security\Policy\Secrets</registry_ignore>
   <registry_ignore type="sregex">\Enum$</registry_ignore>
  </syscheck>

Configuring ignoring files via rules
------------------------------------

> An alternative method to ignore a specific files scanned by syscheck is by using rules and setting the rule level to 0. By doing that the alert will be silenced:

Reviewer 3: ```suggestion
An alternative method to ignore specific files scanned by syscheck is by using rules and setting the rule level to 0. By doing that the alert will be silenced:
```

.. code-block:: xml

  <syscheck>
    <process_priority>19</process_priority>
  </syscheck>

Setting ``process_priority`` value lower than the default, will give syscheck higher priority, more CPU resources and make it run faster. In the example below the nice value for syscheck process is set to minimum:

.. code-block:: xml

  <syscheck>
    <process_priority>-20</process_priority>
  </syscheck>

.. _how_to_fim_database:

Configuring where the database is to be stored
----------------------------------------------

.. versionadded:: 3.12.0

When the Wazuh agent starts it performs a first scan and generates its database. By default, the database is created in disk:

.. code-block:: xml

  <syscheck>
    <database>disk</database>
  </syscheck>

Syscheck can be configured to store the database in memory instead by changing value of the :ref:`database <reference_ossec_syscheck_database>` option:

.. code-block:: xml

  <syscheck>
    <database>memory</database>
  </syscheck>

> The main advantage of using in memory database is the performance as reading and writing data is faster than data stored on disk. The corresponding disadvantage is that the memory must be sufficient to store the data.

Reviewer 3: ```suggestion
The main advantage of using in memory database is the performance, as reading and writing operations are faster than performing them on disk. The corresponding disadvantage is that the memory must be sufficient to store the data.
```
